static/app/views/dashboardsV2/widgetBuilder/widgetBuilder.tsx

```diff
@@ -815,11 +830,17 @@ function WidgetBuilder({
     DisplayType.BIG_NUMBER,
   ].includes(state.displayType);
 
-  const {columns, aggregates, fields} = state.queries[0];
-  const explodedColumns = columns.map(field => explodeField({field}));
-  const explodedAggregates = aggregates.map(field => explodeField({field}));
+  const {columns, aggregates, fields, fieldAliases = []} = state.queries[0];
+
+  const explodedColumns = columns.map((field, index) =>
+    explodeField({field, alias: fieldAliases[index]})
+  );
+  const explodedAggregates = aggregates.map((field, index) =>
+    explodeField({field, alias: fieldAliases[index]})
+  );
```

Comment:
This is going to create new array references on each render, causing rerenders of child components when they could be avoid. If we can use references here, I would advocate for useMemo